src/alembic_utils/pg_policy.py

```diff
@@ -0,0 +1,153 @@
+# pylint: disable=unused-argument,invalid-name,line-too-long
+from __future__ import annotations
+
+from typing import List
+
+from parse import parse
+from sqlalchemy import text as sql_text
+
+from alembic_utils.exceptions import SQLParseFailure
+from alembic_utils.replaceable_entity import ReplaceableEntity, normalize_whitespace
+
+
+class PGPolicy(ReplaceableEntity):
+    """A PostgreSQL Function compatible with `alembic revision --autogenerate`
+
+    **Parameters:**
+
+    * **schema** - *str*: A SQL schema name
+    * **signature** - *str*: A SQL policy name and tablename, separated by "."
+    * **definition** - *str*:  The definition of the policy, incl. permissive, for, to, using, with check
+
+    """
+
+    is_replaceable = False
+
+    @classmethod
+    def from_sql(cls, sql: str) -> PGPolicy:
+        """Create an instance instance from a SQL string"""
+
+        template = "create policy{signature}on{schema}.{tablename}{:s}{definition}"
+        result = parse(template, sql.strip(), case_sensitive=False)
+
+        if result is not None:
+            return cls(
+                schema=result["schema"],
+                signature=f'{result["signature"]}.{result["tablename"]}',
+                definition=result["definition"],
+            )
+        raise SQLParseFailure(f'Failed to parse SQL into PGPolicy """{sql}"""')
+
+    @property
+    def policyname(self):
+        return self.signature.split(".")[0]
+
+    @property
+    def tablename(self):
+        return self.signature.split(".")[1]
+
+    def to_sql_statement_create(self) -> str:
+        """ Generates a SQL "create poicy" statement for PGPolicy """
+
+        return sql_text(
+            f"CREATE POLICY {self.policyname} on {self.literal_schema}.{self.tablename} {self.definition}"
+        )
+
+    def to_sql_statement_drop(self) -> str:
+        """Generates a SQL "drop policy" statement for PGPolicy"""
+
+        return sql_text(f"DROP POLICY {self.policyname} on {self.literal_schema}.{self.tablename}")
+
+    def to_sql_statement_create_or_replace(self) -> str:
+        """Not implemented, postgres policies do not support replace."""
+        raise NotImplementedError()
```

Comment:
populate the to_sql_statement_create_or_replace with a statement that first drops the policy, and then re-creates it